app/models/resource_monitor.rb

```diff
@@ -1,7 +1,6 @@
 class ResourceMonitor < ApplicationRecord
-  validates :service, uniqueness: { scope: %i(resource_type resource_id) },
-                      presence: true
-  validates_presence_of :resource_type, :resource_id
+  validates :app, uniqueness: { scope: :reference }, presence: true
```

Comment:
Don't think it actually matters, but this might be better expressed as "the reference must be unique among the apps" rather than currently where it's saying "the app must be unique among the references".